homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,218 @@
+        self._enasolar = pyenasolar.EnaSolar()
+        self._data: dict[str, Any] = {}
+
+    def _conf_for_inverter_exists(self, serial) -> bool:
+        """Return True if inverter exists in configuration."""
+        return any(
+            entry
+            for entry in self._async_current_entries(include_ignore=False)
+            if serial == entry.unique_id
+        )
+
+    async def _try_connect(self, host):
+        """Needed to mock connection when running tests."""
+        error = None
+        try:
+            await self._enasolar.interogate_inverter(host)
+        except aiohttp.client_exceptions.ClientConnectorError:
+            error = "cannot_connect"
+        except aiohttp.client_exceptions.ClientResponseError:
+            error = "unexpected_response"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            error = "unknown"
+        return error
+
+    def get_serial_no(self):
+        """Needed to mock serial_no when running tests."""
+        return self._enasolar.get_serial_no()
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get options flow for this handler."""
+        return EnaSolarOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Step when user initializes an integration."""
+        _errors = {}
+
+        if user_input is not None:
+            ip_address = await self.hass.async_add_executor_job(
+                _get_ip, user_input[CONF_HOST]
+            )
+            if not ip_address:
+                _errors[CONF_HOST] = "invalid_host"
+            else:
+                self._data[CONF_HOST] = user_input[CONF_HOST]
+                self._data[CONF_NAME] = user_input[CONF_NAME]
+                error = await self._try_connect(self._data[CONF_HOST])
+                if error is not None:
+                    _errors[CONF_HOST] = error
+                else:
+                    if self._conf_for_inverter_exists(self.get_serial_no()):
+                        return self.async_abort(reason="already_configured")
+                    await self.async_set_unique_id(self.get_serial_no())
+                    return await self.async_step_inverter()
+        else:
+            user_input = {}
+            user_input[CONF_NAME] = DEFAULT_NAME
+            user_input[CONF_HOST] = DEFAULT_HOST
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input[CONF_HOST]): cv.string,
+                    vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): cv.string,
+                }
+            ),
+            errors=_errors,
+            last_step=False,
+        )
+
+    async def async_step_inverter(self, user_input=None):
+        """Give the user the opportunities to override inverter config."""
```

Comment:
Should this not move to the option flow instead? Needed to setup the integration?

Reply:
No, the basic configuration of the inverter is fixed, if the hardware is modified, the integration should be deleted and re-added. Only the polling window is able to be updated once the inverter is added.